readme.txt
==== readme.txt ====
Updated December 28, 2014

=== Contents ===
* Items of Note
* F.A.Q.
* Internal Code Conventions
* Review Tickets

=== Items of Note ===
* This Theme does not use Custom Header Images
* Posts without titles can be reached by clicking on the linked 'Comment' reference
* Posts without titles and password protected will not have a direct link to the single view of the post.
* Author email link will only appear on 'author.php' rendered pages if their website URL has been added to their profile.

=== F.A.Q. ===
Q: Why is the theme not using the WordPress core search form since Ground Floor version 2.1?
The update routine for WordPress themes overwrites the existing files with the files found in the updated version of the theme. If a file is removed in an update it must be manually removed from the website afterward. This does not affect new (re-)installation of the theme.

Q: Why are not all of the archive posts showing an excerpt only?
The archive templates are created to show as "mullet" loops where the first few posts will show in their entirety then the following posts will show as excerpts.

=== Internal Code Conventions ===
* Unique function prefix: `gf_`
* Expected theme slug: `ground-floor`
* Theme textdomain is expected to be the same as the theme slug: `ground-floor`

=== Review Tickets ===
* http://themes.trac.wordpress.org/ticket/278
* http://themes.trac.wordpress.org/ticket/1363
* http://themes.trac.wordpress.org/ticket/3374
* http://themes.trac.wordpress.org/ticket/4165 - version 1.7.1
* http://themes.trac.wordpress.org/ticket/5143 - version 1.8
* http://themes.trac.wordpress.org/ticket/7445 - version 1.9
* http://themes.trac.wordpress.org/ticket/7446 - version 1.9.1
* http://themes.trac.wordpress.org/ticket/8516 - version 2.0
* http://themes.trac.wordpress.org/ticket/10292 - version 2.1
* http://themes.trac.wordpress.org/ticket/11663 - version 2.2 - March 2013
* http://themes.trac.wordpress.org/ticket/13433 - version 2.2.1 - July 2013
* http://themes.trac.wordpress.org/ticket/15297 - version 2.2.2 - November 2013
* https://themes.trac.wordpress.org/ticket/17977 - version 2.2.3 - April 2014
* https://themes.trac.wordpress.org/ticket/21491 - version 2.3 - November 2014
* https://themes.trac.wordpress.org/ticket/22352 - version 2.3.2 - December 2014

changelog.txt
=== Ground Floor ===
== Version 2.3.2 ==
Released December 2014
= Code =
* Added support for `add_theme_support( 'title-tag' )`
* Added support for `BNS Login` using dashicons
* Code formatting to better meet WordPress Coding Standards
* Miscellaneous inline documentation updates
* Moved `$content_width` into callback function

= CSS =
* Added compatibility styles for `BNS Theme Details`

= Miscellaneous =
* Corrections to `readme.txt` to address mark-up issues

/** ------------------------------------------------------------------------- */
== Version 2.3.1.1 ==
* Additional reversion updates

== Version 2.3.1 ==
* Reverted name change: `gf_setup` -> `ground_floor_setup`

/** ------------------------------------------------------------------------- */
== Version 2.3 ==
= Code =
* Re-factored `comment_add_userid` to note various user roles as comment classes
* Minor i18n corrections and updates
* Removed `wp_list_bookmarks` call from the default sidebar widgets
* Removed micro-ID class insertion into comment classes
* Renamed `ground_floor_setup` to `gf_setup` to be more consistent

= CSS =
* Ensured embedded "tweets" are set to 100% width in the content area

= Miscellaneous =
* Set i18n textdomain references to match theme slug
* Updated `readme.txt` file with better details about the theme

/** ------------------------------------------------------------------------- */
== Version 2.2.3 ==
= Code =
* Reformatted code to better meet WordPress Coding Standards

= CSS =
* Added `address` definition
* Fixed two minor typos and corrected `.wp-caption` alignment issues

= Miscellaneous =
* Added `Textdomain` to theme header block
* Added FAQ: Why are not all of the archive posts showing an excerpt only?
* Changed theme tag `fixed-width` to `fixed-layout`
* Replace `screenshot.png` with a 880 x 660 pixel version
* Update Copyright years

/** ------------------------------------------------------------------------- */
== Version 2.2.2 ==
= Code =
* Removed unused variables and parameters as part of code clean-up

= CSS =
* Added some basic list element styles for sub-elements
* Added some minor aesthetic changes to some elements
* Added `word-wrap` properties for pages and posts
* Added better list element styles
* Basic clean up of old and/or duplicated properties

/** ------------------------------------------------------------------------- */
== Version 2.2.1 ==
= Code =
* Added constant 'GF_HOME_URL'
* Applied 'GF_HOME_URL' as appropriate
* Change Featured Image to centered and full size from left aligned thumbnail
* General code cleanup
* Simplified custom menu support code

= CSS =
* Added 'display: none;' for `screen reader text` labels
* Adjusted galleries CSS to provide a larger thumbnail view where space is available
* Adjusted `img` properties related to the Featured Image
* Keep Featured Images to width of container with `wp-post-image` properties

/** ------------------------------------------------------------------------- */
== Version 2.2 ==
= Code =
* Added 'search' results template
* Dropped the "subscribe" link from the post meta in the 'single' template
* Refactored code formatting and code block termination comments
* Refactored widget area definitions into `gf_widgets` function and hooked into `widget_init`
* Refactored to be more i18n compatible
* Removed 'searchform.php' in favor of using WordPress core version
* Updated DOCTYPE and other related header elements

= CSS =
* Changed size of archive titles to a much smaller more aesthetic size
* Change search input field width in sidebar (see 'searchform.php' removal)

= Miscellaneous
* Change 'style.css' header texts to reflect core trac ticket #16868
* Updated copyright year
* Updated 'readme' with new F.A.Q. section and answer regarding the removal of the theme version of 'searchform.php'

/** ------------------------------------------------------------------------- */
== Version 2.1 ==
Changelog: December 9, 2012
= Code =
* Added filters to allow the theme version text to be completely overwritten
* Added 'Ground Floor Custom Background Callback' to specifically set the CSS `background-attachment` property to `fixed` by default.
* Addressed conditional to display threaded comments if they are open or closed
* Fixed spacing issue in meta details if post has no tags
* Refactored sidebar definitions to include names and descriptions
* Reworded the theme version texts
* Removed duplicate `content width` entry
* Removed `gf_login` function in favor of using its parent plugin BNS Login
* Removed `gf_modified_post` as unused code (may re-introduce in a later version)

= CSS =
* Adjusted search box input field to fill width
* Change `font-size` properties to percentage based values with base font-size set to 100%
* Fixed aesthetics of the menu for the current page
* Hid search buttons by un-commenting the `hidden` class properties

= Miscellaneous =
* Minor code reformatting
* Documentation updates and corrections
* More complete PHPDocs style documentation added

/** ------------------------------------------------------------------------- */
== Version 2.0 ==
Changelog: July 6, 2012
= Code =
* Addressed `_e` and other output / i18n issues
* Changed TEMPLATEPATH to `get_template_directory()`
* Refactored `wp_title` usage
* Refactored script enqueue of threaded comments
* Removed code for call to deprecated function `add_custom_background`
* Removed code for call to deprecated function `get_theme_data`
* Removed `function_exists( 'dynamic-sidebar' )` checks - should not be needed
* Updates to i18n support

= Miscellaneous =
* Various inline documentation updates and corrections
* Dropped widget-map image file from package

/** ------------------------------------------------------------------------- */
== Version 1.9 ==
Changelog: April 19, 2012
= Code =
* Added conditionals to display website (with email) and biography only if information has been entered into the user fields

author.php
<?php
/**
 * Author Template
 * Displays author page is required to be displayed
 *
 * @package     GroundFloor
 * @since       1.0
 *
 * @link        http://buynowshop.com/themes/ground-floor/
 * @link        https://github.com/Cais/ground-floor/
 * @link        http://wordpress.org/extend/themes/ground-floor/
 *
 * @author      Edward Caissie <edward.caissie@gmail.com>
 * @copyright   Copyright (c) 2009-2014, Edward Caissie
 *
 * @version     2.2
 * @date        March 11, 2013
 * Refactored code formatting and code block termination comments
 * Refactored post meta to be more i18n compatible
 *
 * @version     2.3
 * @date        November 1, 2014
 * Minor i18n corrections and updates
 */

get_header();

/** Set the $curauth variable */
$curauth = ( get_query_var( 'author_name ' ) ) ? get_user_by( 'id', get_query_var( 'author_name' ) ) : get_userdata( get_query_var( 'author' ) ); ?>

	<div id="main-blog">

		<div id="content">

			<div id="author" class="<?php
			/** Add class as related to the user role (see 'Role:' drop-down in User options) */
			if ( user_can( $curauth->ID, 'administrator' ) ) {
				echo 'administrator';
			} elseif ( user_can( $curauth->ID, 'editor' ) ) {
				echo 'editor';
			} elseif ( user_can( $curauth->ID, 'contributor' ) ) {
				echo 'contributor';
			} elseif ( user_can( $curauth->ID, 'subscriber' ) ) {
				echo 'subscriber';
			} else {
				echo 'guest';
			}
			/** End if - user can */

			/** Add user classes by ID */
			/** First user/administrator */
			if ( ( $curauth->ID ) == '1' ) {
				echo ' administrator-prime';
			}
			/** End if - current author ID is 1 */
			/** Additional Example */
			/** elseif ( ( $curauth->ID ) == '2' ) echo ' user-id-2 jellybeen'; */
			?>">

				<h2>
					<?php printf( __( 'About %1$s', 'ground-floor' ), $curauth->display_name ); ?>
				</h2>

				<ul>
					<?php if ( ! empty( $curauth->user_url ) ) { ?>
						<li><?php printf(
								__( 'Website: %1$s or %2$s', 'ground-floor' ),
								'<a href="' . $curauth->user_url . '">' . $curauth->user_url . '</a>',
								'<a href="mailto:' . $curauth->user_email . '">' . __( 'email', 'ground-floor' ) . '</a>'
							); ?>
						</li>
					<?php
					}
					/** End if - not empty current author - url */
					if ( ! empty( $curauth->user_description ) ) {
						?>
						<li><?php printf( __( 'Biography: %1$s', 'ground-floor' ), $curauth->user_description ); ?></li>
					<?php } /** end if - not empty current author - description */ ?>
				</ul>

			</div>
			<!-- #author -->

			<h2>
				<?php printf( __( 'Posts by %1$s:', 'ground-floor' ), $curauth->display_name ); ?>
			</h2>

			<!-- Start the_Loop -->
			<?php
			if ( have_posts() ) {
				$count = 0;

				while ( have_posts() ) {
					the_post();
					$count ++; ?>

					<div <?php post_class(); ?> id="post-<?php the_ID(); ?>">
						<h2>
							<a href="<?php the_permalink() ?>" rel="bookmark" title="<?php _e( 'Permanent Link to', 'ground-floor' ); ?> <?php the_title_attribute(); ?>"><?php the_title(); ?></a>
						</h2>

						<div class="post-details">

							<?php printf( __( '%1$s on %2$s', 'ground-floor' ), gf_use_posted(), get_the_time( get_option( 'date_format' ) ) );
							if ( ! post_password_required() ) {
								/** Hide Comment(s) if password required to read post */
								echo ' ';
								comments_popup_link( __( 'with No Comments', 'ground-floor' ), __( 'with 1 Comment', 'ground-floor' ), __( 'with % Comments', 'ground-floor' ), '', __( '(Comments Closed)', 'ground-floor' ) );
							}
							/** password protected post test */
							edit_post_link( __( 'Edit', 'ground-floor' ), __( ' &#124; ', 'ground-floor' ), __( '', 'ground-floor' ) );
							_e( '<br />in ', 'ground-floor' ); ?><?php the_category( ', ' ) ?>
							<br />
							<?php the_tags( __( 'as ', 'ground-floor' ), ', ', '' ); ?>
							<br />

						</div>
						<!-- .post-details -->

						<?php
						if ( $count == 1 ) {
							the_content();
							wp_link_pages(
								array(
									'before'         => '<p><strong>' . __( 'Pages:', 'ground-floor' ) . '</strong> ',
									'after'          => '</p>',
									'next_or_number' => 'number'
								)
							);
						} else {
							the_excerpt();
						} /** End if - count is 1 */
						?>

						<div class="clear"></div>
						<!-- For inserted media at the end of the post -->

					</div><!-- post_class -->

				<?php } /** end while - have posts */ ?>

				<div id="nav-global" class="navigation">
					<div class="left">
						<?php next_posts_link( __( '&laquo; Previous entries ', 'ground-floor' ) ); ?>
					</div>
					<div class="right">
						<?php previous_posts_link( __( ' Next entries &raquo;', 'ground-floor' ) ); ?>
					</div>
				</div><!-- .navigation -->

				<div class="clear"></div>

			<?php } else { ?>

				<h2>
					<?php printf( __( 'Search Results for: %s', 'ground-floor' ), '<span>' . esc_html( get_search_query() ) . '</span>' ); ?>
				</h2>
				<p><?php _e( 'Sorry, there are no posts by this author.', 'ground-floor' ); ?></p>

				<?php
				get_search_form();
			} ?>
			<!-- end the Loop -->

		</div>
		<!-- #content -->

	</div><!-- #main-blog -->

<?php
get_sidebar();
get_footer();